addons/l10n_it/migrations/15.0.0.3/post-migrate.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+from odoo import api, SUPERUSER_ID
+
+def migrate(cr, version):
+
+    cr.execute("""
+        INSERT INTO account_account_account_tag
+        SELECT a.id, att.account_account_tag_id
+        FROM account_account AS a
+        JOIN account_account_template AS at ON a.code LIKE CONCAT(at.code, '%')
+        JOIN account_account_template_account_tag AS att ON at.id = att.account_account_template_id
+        JOIN ir_model_data AS imd ON imd.res_id = a.id
```

Comment:
Wrong. But I nearly didn't see it because of your aliases :p 

You're joining on account.account instead of joining on account.account.template. It means old dbs (pre-9.0) will not see their tags updated (since their account.accounts don't have an xmlid), as well as chart of accounts exactly identical to yours, but created by hand. 

Note that you'll then also need to add a condition on the company to which the accounts belong: the account_fiscal_country_id must be Italy. If we don't do that and apply the change I was just explaining, you'll also set the tags in any foreign company with similar account codes (this could happen by chance)